fix component double-parsing (fixes #912)

ngokevin opened the pull request:
This is blocking adding a property type for assets and the model work:

- Need to change Component to not call `parse` before doing `buildData` since that already does schema parsing. We used to call it because we needed the `style-attr`-parsing. Since we only need that, I changed `buildData` to do `style-attr` parsing for multi-prop schemas.
- Simplified Component.parse/Component.stringify to use the schema functions since it was doing the same thing the schema already handles.
- Bind property parse/stringify to component prop definition to give vec3 parser access to default value

dmarcos commented on src/core/component.js:
```diff
@@ -111,63 +112,27 @@ Component.prototype = {
-      if (!schema[key]) { return; }
-      processedData[key] = schema[key].stringify(processedData[key]);
-    }
+    if (isSingleProp(schema)) { return stringifyProperty(data, schema); }
```
There was an error returned here if the type was not a vaid one. If I do a type on a type name does it now fail silently?

dmarcos replied:
I think if a schema specifies a ```type``` and the type is not a valid one we should return a loud error somewhere. I see people declaring a type as ```v3``` or ```vector3``` and not knowing how it's not working.

ngokevin replied:
the schema module does property type lookup and warnings during the processing stage.

ngokevin replied:
that already happens at https://github.com/aframevr/aframe/blob/dev/src/core/schema.js#L63

dmarcos commented on src/core/schema.js:
```diff
@@ -59,43 +60,59 @@ function processPropertyDefinition (propDefinition) {
   propType = propertyTypes[typeName];
-
   if (!propType) {
     warn('Unknown property type: ' + typeName);
```
This should be an error instead of a warning.

dmarcos commented on src/core/schema.js:
```diff
@@ -59,43 +60,59 @@ function processPropertyDefinition (propDefinition) {
   if (!propType) {
-    warn('Unknown property type: ' + typeName);
+    error('Unknown property type: ' + typeName);
     return propDefinition;
```
Do we still need the return if we trigger an error?

ngokevin replied:
ok changed to throw rather than log

dmarcos commented:
travis is complaining

dmarcos commented:
This one is ready to go if we just fix the tests

ngokevin commented:
yeah trying to figure out a failure

ngokevin commented:
Okay, updated. The component attribute animation was running into `undefined` for the component property value. Also removed a test involving `removeAttribute(component, propName, propValue)` since that doesn't seem necessary.
